RetailCoder.VBE/UI/Command/Refactorings/FormDesignerRefactorRenameCommand.cs

```diff
@@ -45,35 +45,39 @@ protected override void OnExecute(object parameter)
             }
         }
 
-        private Declaration GetTarget()
+        private Declaration GetTarget(QualifiedModuleName? qualifiedModuleName = null)
         {
-            var project = _vbe.ActiveVBProject;
-            var component = _vbe.SelectedVBComponent;
+            (var projectId, var component) = 
+                qualifiedModuleName.HasValue 
+                                   ? (qualifiedModuleName.Value.ProjectId, qualifiedModuleName.Value.Component)
+                                   : (Vbe.ActiveVBProject.ProjectId, Vbe.SelectedVBComponent);
+                        
+            if (component?.HasDesigner ?? false)
             {
-                if (Vbe.SelectedVBComponent != null && Vbe.SelectedVBComponent.HasDesigner)
+                if (qualifiedModuleName.HasValue)
                 {
-                    var designer = ((dynamic)component.Target).Designer;
-
-                    if (designer.selected.count == 1)
-                    {
-                        var control = designer.selected.item(0);
-                        var result = _state.AllUserDeclarations
-                            .FirstOrDefault(item => item.DeclarationType == DeclarationType.Control
-                                                    && project.HelpFile == item.ProjectId
-                                                    && item.ComponentName == component.Name
-                                                    && item.IdentifierName == control.Name);
-
-                        Marshal.ReleaseComObject(control);
-                        Marshal.ReleaseComObject(designer);
-                        return result;
-                    } else {
-                        var message = string.Format(RubberduckUI.RenameDialog_AmbiguousSelection);
-                        _messageBox.Show(message, RubberduckUI.RenameDialog_Caption, MessageBoxButtons.OK,
-                            MessageBoxIcon.Exclamation);
-                    }
+                    return _state.DeclarationFinder
+                                 .MatchName(qualifiedModuleName.Value.Name)
+                                 .SingleOrDefault(m => m.ProjectId == projectId
```

Comment:
I do not think that this is what you want to return. This will just return the component, not the selected control.

Reply:
This is by design - if no control is selected, then the form is selected; the rename will then act against the form.